src/main/java/io/anserini/rerank/RerankerContext.java

```diff
@@ -46,6 +47,20 @@ public RerankerContext(IndexSearcher searcher, K queryId, Query query, String qu
     this.queryTokens = queryTokens;
     this.filter = filter;
     this.searchArgs = searchArgs;
+    this.weightedQueryTokens = null;
+  }
+
+  public RerankerContext(IndexSearcher searcher, K queryId, Query query, String queryDocId, String queryText,
```

Comment:
hi @AileenLin can you explain why we need to add this code path?

`SearchCollection` is able to perform Rocchio/RM3 with impact exists, right?
E.g., https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SearchCollection.java#L1152

Reply:
the SimpleImpact search method take in a Map of token frequency, we can either add the weitedQueryToken here, or add another search method to process text in SimpleImpact. But if for pretokenized Map of token frequency, I prefered the second way to do it

Reply:
In that case, would it make sense for `SearchCollection` and `SimpleImpactSearcher` to use the same code path?